hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapEventPublishingService.java

```diff
@@ -0,0 +1,134 @@
+import com.hazelcast.map.impl.event.EntryEventData;
+import com.hazelcast.map.impl.event.EventData;
+import com.hazelcast.monitor.impl.LocalReplicatedMapStatsImpl;
+import com.hazelcast.replicatedmap.ReplicatedMapCantBeCreatedOnLiteMemberException;
+import com.hazelcast.replicatedmap.impl.record.AbstractReplicatedRecordStore;
+import com.hazelcast.replicatedmap.impl.record.ReplicatedRecordStore;
+import com.hazelcast.spi.EventFilter;
+import com.hazelcast.spi.EventPublishingService;
+import com.hazelcast.spi.EventRegistration;
+import com.hazelcast.spi.EventService;
+import com.hazelcast.spi.NodeEngine;
+import java.util.EventListener;
+
+import static com.hazelcast.replicatedmap.impl.ReplicatedMapService.SERVICE_NAME;
+
+/**
+ * Dispatches published events on replicated map to corresponding listeners.
+ */
+public class ReplicatedMapEventPublishingService implements EventPublishingService {
+    private final ReplicatedMapService replicatedMapService;
+    private final NodeEngine nodeEngine;
+    private final Config config;
+    private final EventService eventService;
+
+    public ReplicatedMapEventPublishingService(ReplicatedMapService replicatedMapService) {
+        this.replicatedMapService = replicatedMapService;
+        this.nodeEngine = replicatedMapService.getNodeEngine();
+        this.config = nodeEngine.getConfig();
+        this.eventService = nodeEngine.getEventService();
+    }
+
+    @Override
+    public void dispatchEvent(Object event, Object listener) {
+        if (nodeEngine.getConfig().isLiteMember()) {
+            final Class<?> eventClazz = event.getClass();
+            nodeEngine.getLogger(ReplicatedMapService.class).warning("Replicated map event ignored by lite member " + eventClazz);
+            return;
+        }
+
+        if (!(event instanceof EntryEventData)) {
+            return;
+        }
+        EntryEventData entryEventData = (EntryEventData) event;
+        Member member = getMember(entryEventData);
+        EntryEvent entryEvent = createDataAwareEntryEvent(entryEventData, member);
+        EntryListener entryListener = (EntryListener) listener;
+        switch (entryEvent.getEventType()) {
+            case ADDED:
+                entryListener.entryAdded(entryEvent);
+                break;
+            case EVICTED:
+                entryListener.entryEvicted(entryEvent);
+                break;
+            case UPDATED:
+                entryListener.entryUpdated(entryEvent);
+                break;
+            case REMOVED:
+                entryListener.entryRemoved(entryEvent);
+                break;
+            // TODO handle evictAll and clearAll event
+            default:
+                throw new IllegalArgumentException("event type " + entryEvent.getEventType() + " not supported");
+        }
+        String mapName = ((EntryEventData) event).getMapName();
+        if (config.findReplicatedMapConfig(mapName).isStatisticsEnabled()) {
```

Comment:
if a wildcard used in the repl-map's name, this lookup becomes very costly (pattern matching). do we have any other choice here ?